### PR TITLE
Ignore page view event if tag id is GA4

### PR DESCRIPTION
**Pull request by jivey**
for: https://github.com/openstax/unified/issues/2225

Probably don't want to release this until https://github.com/openstax/unified/issues/2226 is figured out.

### CODE REVIEW

File: src/gateways/googleAnalyticsClient.ts [hunk 1]
```diff
@@ -196,18 +213,20 @@ class GoogleAnalyticsClient {
       return;
     }
 
-    for (const tagId of this.tagIds) {
+    for (const tag of this.tags) {
       command.payload = {...command.payload, queue_time: queueTime };
+
       if (command.name === 'event') {
-        this.gtag('event', command.eventName, { ...command.payload, send_to: tagId });
+        if (tag.ignoredEventNames.includes(command.eventName)) { return; }
```
Comment (Dantemss) [inline src/gateways/googleAnalyticsClient.ts:220]: I think this should be `continue` instead?

**Review by Dantemss — CHANGES_REQUESTED** [review]

File: src/gateways/googleAnalyticsClient.ts (hunk 1 above)
Comment (Dantemss) [reply src/gateways/googleAnalyticsClient.ts:220]: But then maybe you need some code to abort if everything got skipped

Comment (Dantemss) [reply src/gateways/googleAnalyticsClient.ts:220]: Or maybe no need to check for the abort condition, since there's nothing after the loop.

Comment (jivey) [reply src/gateways/googleAnalyticsClient.ts]: Whoops good catch, that was definitely a typo and should have been `continue`

File: src/gateways/googleAnalyticsClient.ts [hunk 2]
```diff
@@ -196,18 +213,20 @@ class GoogleAnalyticsClient {
       return;
     }
 
-    for (const tagId of this.tagIds) {
+    for (const tag of this.tags) {
       command.payload = {...command.payload, queue_time: queueTime };
+
       if (command.name === 'event') {
-        this.gtag('event', command.eventName, { ...command.payload, send_to: tagId });
+        if (tag.ignoredEventNames.includes(command.eventName)) { continue; }
```
Comment (TomWoodward) [inline src/gateways/googleAnalyticsClient.ts:220]: you could invert the condition and put the gtag call inside it, that feels a bit nicer than using the continue to short circuit

**Review by TomWoodward — CHANGES_REQUESTED** [review]

**Comment by TomWoodward** [comment]
> Probably don't want to release this until https://github.com/openstax/unified/issues/2226 is figured out.

i don't think we have to wait, ga4 reporting won't be significantly more broken than it is already

**Review by TomWoodward — APPROVED** [review]

**Review by Dantemss — APPROVED** [review]

**Comment by jivey** [comment]
@Malar-Natarajan I've also temporarily added ga3 and ga4 testing ids to this PR. For testing, we want to make sure page_view events get sent to ga3 but not ga4.

**Comment by jivey** [comment]
@Malar-Natarajan I forgot to also mention that in the GA4 debug view, it's possible that the automatic, enhanced tracking pageview event will still be there, which we expect. You can distinguish our custom pageview event, which should not be there, by it having a `queue_time` param:

**Comment by Malar-Natarajan** [comment]
> in the GA4 debug view, it's possible that the automatic, enhanced tracking pageview event will still be there, which we expect. You can dis

ok, so this is why there are 2 page views

**Pull request merged**